packages/flutter_tools/gradle/src/main/groovy/flutter.groovy

```diff
@@ -0,0 +1,1293 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import static groovy.io.FileType.FILES
+
+import com.android.build.OutputFile
+import groovy.json.JsonSlurper
+import java.nio.file.Path
+import java.nio.file.Paths
+import java.util.regex.Matcher
+import java.util.regex.Pattern
+import org.apache.tools.ant.taskdefs.condition.Os
+import org.gradle.api.DefaultTask
+import org.gradle.api.GradleException
+import org.gradle.api.Project
+import org.gradle.api.Plugin
+import org.gradle.api.Task
+import org.gradle.api.file.CopySpec
+import org.gradle.api.file.FileCollection
+import org.gradle.api.logging.LogLevel
+import org.gradle.api.tasks.Copy
+import org.gradle.api.tasks.InputFiles
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.Internal
+import org.gradle.api.tasks.OutputDirectory
+import org.gradle.api.tasks.OutputFiles
+import org.gradle.api.tasks.Optional
+import org.gradle.api.tasks.TaskAction
+import org.gradle.api.tasks.bundling.Jar
+import org.gradle.internal.os.OperatingSystem
+import org.gradle.util.VersionNumber
+
+/** For apps only. Provides the flutter extension used in app/build.gradle.
+ * The versions specified here should match the values used in
+ * ../lib/src/android/gradle_utils.dart, so when bumping, make sure to update
+ * the versions specified there.
+*/
+class FlutterExtension {
+    /** Sets the compileSdkVersion used by default in Flutter app projects. */
+    static int compileSdkVersion = 33
+
+    /** Sets the minSdkVersion used by default in Flutter app projects. */
+    static int minSdkVersion = 16
+
+    /** Sets the targetSdkVersion used by default in Flutter app projects. */
+    static int targetSdkVersion = 33
+
+    /**
+     * Sets the ndkVersion used by default in Flutter app projects.
+     * Chosen as default version of the AGP version below as found in
+     * https://developer.android.com/studio/projects/install-ndk#default-ndk-per-agp
+     */
+    static String ndkVersion = "23.1.7779620"
+
+    /**
+     * Specifies the relative directory to the Flutter project directory.
+     * In an app project, this is ../.. since the app's build.gradle is under android/app.
+     */
+    String source
+
+    /** Allows to override the target file. Otherwise, the target is lib/main.dart. */
+    String target
+}
+
+// This buildscript block exists solely for the compatibility with projects that
+// have not migrated to the declarative apply of the Flutter Gradle Plugin.
+buildscript {
```

Comment:
Can this block remain in the old `flutter.gradle` file? Seems like that'd be cleanest, if it's not meant to apply to projects that aren't using the old file.

Reply:
Yeah I tried doing that, but then the build fails with:

```
 What went wrong:
Could not compile script '/Users/bartek/fvm/versions/master/packages/flutter_tools/gradle/src/main/groovy/flutter.groovy'.
> startup failed:
  script '/Users/bartek/fvm/versions/master/packages/flutter_tools/gradle/src/main/groovy/flutter.groovy': 7: unable to resolve class com.android.build.OutputFile
   @ line 7, column 1.
     import com.android.build.OutputFile
     ^

  1 error
```

So `buildscript` applies only to the buildscript it was called on. `apply`-ed script plugins don't inherit its dependencies.

Reply:
Hmm, I see. But that error only happens from a project that's using the old style?

Also, gosh that's wacky that this block down here (after not only all the imports, but also a class definition) can affect the behavior of the imports. Especially given that this is in Groovy, a *scripting* language, where one ordinarily expects even the top-level definitions to behave as imperative statements that get executed in order.

Anyway, maybe just make that explicit, then. Like:
```suggestion
// This buildscript block supplies dependencies for this file's own import
// declarations above. It exists solely for compatibility with projects that
// have not migrated to declaratively apply the Flutter Gradle Plugin;
// for those that have, FGP's `build.gradle.kts`  takes care of this.
buildscript {
```